### ISSUE TITLE
The test `TestDatabaseConnection` is flaky

### ISSUE THREAD

**Issue by rh-hemartin** [kubearchive/kubearchive#382]
Sometimes it fails and sometimes passes. I think the problem is with the logs of Kubernetes, which seem not to refresh when a new container is created. For example in https://github.com/kubearchive/kubearchive/actions/runs/10723346226/job/29736419010?pr=378 you can see that after 10 retries with jitter, which take approx 50 seconds, the sink did not try to connect to the database a second time. That looks like the log, for whatever reason, is not refreshed.

**Comment by rh-hemartin** [comment]
I think we could solve the race condition of logs by not upscaling and downscaling our deployments during testing. I think the goal of the test is to check that the retry operation is occuring, in other words we are testing that our usage of the retry library is correct.

When I first encountered this test I tried to unit test the underlying function instead, but it is difficult because of how its designed. The function is mostly glue, with not much logic, so I would try to refactor the test to make it easier to test.
